### PR TITLE
Allow for case-insensitive package names

### PR DESCRIPTION
**Pull request by mdrohmann**
https://www.pivotaltracker.com/story/show/175469208

### CODE REVIEW

File: internal/testhelpers/e2e/session.go [hunk 1]
```diff
@@ -193,6 +193,7 @@ func (s *Session) SpawnCmdWithOpts(exe string, opts ...SpawnOptions) *termtest.C
 			RetainWorkDir:  true,
 			ObserveExpect:  observeExpectFn(s),
 			ObserveSend:    observeSendFn(s),
+			// ExtraOpts:      []expect.ConsoleOpt{expect.WithStdout(os.Stdout)},
```
Comment (MDrakos) [inline internal/testhelpers/e2e/session.go:196]: Let's leave a comment here to explain why this code is commented out and why you may want to enable it

File: pkg/platform/model/inventory.go [hunk 2]
```diff
@@ -50,25 +52,59 @@ func IngredientByNameAndVersion(language, name, version string, prefix Namespace
...
+						})
 				}
 			}
 		}
 	}
 
-	return nil, locale.NewInputError("inventory_ingredient_version_not_available", "Version {{.V0}} is not available for package {{.V1}} on the ActiveState Platform", version, name)
+	bestMatch, err := FilterForBestIngredientMatch(candidates, name)
+	if err != nil {
+		return nil, locale.NewInputError("inventory_ingredient_version_not_available", "Version {{.V0}} is not available for package {{.V1}} on the ActiveState Platform", version, name)
```
Comment (MDrakos) [inline pkg/platform/model/inventory.go:78]: The error from `FilterForBestIngredientMatch` is being thrown away here. Shouldn't we just return the error, or wrap again?

File: pkg/platform/model/inventory.go [hunk 3]
```diff
@@ -50,25 +52,59 @@ func IngredientByNameAndVersion(language, name, version string, prefix Namespace
...
+		for _, feature := range c.LatestVersion.ProvidedFeatures {
+			if *feature.Feature == name {
+				return c, nil
+			}
+		}
+	}
+
+	if len(candidates) == 0 {
+		return nil, locale.NewInputError("inventory_ingredient_no_version_available", "No versions are available for package {{.V0}} on the ActiveState Platform", name)
+	}
```
Comment (MDrakos) [inline pkg/platform/model/inventory.go:97]: nit: I think this check makes more sense at the beginning of this function

File: pkg/platform/model/inventory.go [hunk 4]
```diff
@@ -82,21 +118,33 @@ func IngredientWithLatestVersion(language, name string, prefix NamespacePrefix)
...
+		return nil, errs.Wrap(err, "Could not retrieve a match for ingredient %s.", name)
+	}
 
-	return nil, locale.NewInputError("inventory_ingredient_no_version_available", "No versions are available for package {{.V0}} on the ActiveState Platform", name)
+	latest := &IngredientAndVersion{
+		bestMatch.V1SearchIngredientsResponseIngredientsItems,
+		*bestMatch.LatestVersion.Version,
+		bestMatch.Namespace,
+	}
+	return latest, nil
```
Comment (MDrakos) [inline pkg/platform/model/inventory.go:147]: Can't we just return the `bestMatch` at this point?

**Review by MDrakos — CHANGES_REQUESTED** [review]
Looks good! Just a few minor questions/concerns

File: pkg/platform/model/inventory.go (hunk 2 above)
Comment (mdrohmann) [reply pkg/platform/model/inventory.go:78]: thanks for catching that.

File: pkg/platform/model/inventory.go (hunk 4 above)
Comment (mdrohmann) [reply pkg/platform/model/inventory.go:147]: yes, indeed!

**Review by MDrakos — APPROVED** [review]

**Pull request merged**